armi/apps.py

```diff
@@ -221,6 +229,76 @@ def getParamRenames(self) -> Dict[str, str]:
             self._paramRenames = renames, self._pm.counter
         return renames
 
+    def registerUserPlugins(self, pluginPaths):
+        """
+        Register additional plugins passed in by importable paths.
+        These plugins may be provided e.g. by an application during startup
+        based on user input.
+        Format expected to be a list of full namespaces to plugin classes.
+        There should be a comma between individual plugins and dots representing
+        the importable python namespace. e.g.
+
+        ``armi.stuff.plugindir.pluginMod.pluginCls,armi.whatever.plugMod2.plugCls2``.
+        or on Linux/Unix:
+        ``/path/to/pluginMod.py:pluginCls,/path/to/plugMod2.py:plugCls2```,
+        or on Windows:
+        ``C:\\\\path\\to\\pluginMod.py:pluginCls,C:\\\\path\\to\\plugMod2.py:plugCls2```,
+
+        Notes
+        -----
+        These paths are meant to be taken from a settings file, though this method
+        is public. The idea is that these "user plugins" differ from regular plugins
+        because they are defined during run time, not import time. As such, we
+        restrict their flexibility and power as compared to the usual ArmiPlugins.
+        """
+        self.__initNewPlugins()
+
+        for pluginPath in pluginPaths:
+            if sep in pluginPath:
+                # The path is of the form: /path/to/why.py:MyPlugin
+                self.__registerUserPluginsAbsPath(pluginPath)
+            else:
+                # The path is of the form: armi.thing.what.MyPlugin
+                self.__registerUserPluginsInternalImport(pluginPath)
+
+    def __registerUserPluginsAbsPath(self, pluginPath):
+        """Helper method to register a single UserPlugin where
+        the given path is of the form: /path/to/why.py:MyPlugin
+        """
+        # determine if we have that Windows file path (C:\\path\to\whatever.py)
+        isWindows = False
+        if ":\\" in pluginPath:
```

Comment:
**TODO**: JOHN!

I just found out that most Windows machines use the format: `C:\whatever`, but it's only the OLD versions of Windows that use `C:\\whatever`.

So, apparently, to support Windows, you need to support both. So, joy.

Reply:
Fixed